net/respondd/src/respondd.c

```diff
@@ -165,6 +183,60 @@ static const struct respondd_provider_info * get_providers(const char *filename)
 	return ret;
 }
 
+bool schedule_push_request(struct request_schedule *s, struct request_task *new_task) {
+	if (s->length >= SCHEDULE_LEN)
+		// schedule is full
+		return false;
+
+	if (!s->list_head || s->list_head->scheduled_time > new_task->scheduled_time) {
+		new_task->next = s->list_head;
+		s->list_head = new_task;
+	} else {
+		struct request_task *t;
+		for (t = s->list_head; t && t->next; t = t->next) {
+			if (t->next->scheduled_time > new_task->scheduled_time) {
+				break;
+			}
+		}
+		new_task->next = t->next;
+		t->next = new_task;
+	}
+
+	s->length++;
+	return true;
+}
+
+int64_t schedule_idle_time(struct request_schedule *s) {
+	if (!s->list_head)
+		// nothing to do yet, wait nearly infinite time
+		return 3600000;
```

Comment:
This should be `return 0;`, which according to `socket(7)` means "never timeout".